src/shims/unix/fd.rs

```diff
@@ -58,6 +60,19 @@ pub trait FileDescription: std::fmt::Debug + Any {
         // so we use a default impl here.
         false
     }
+    // Check the readiness of the file description support to be epolled and update the ready list.
+    fn check_and_update_readiness<'tcx>(&self, ecx: &mut MiriInterpCx<'tcx>) -> InterpResult<'tcx> {
```

Comment:
move this into a free method or a MiriInterpCx extension method, it's only called from epoll code, not general file descriptor code, and it should never be overwritten

Reply:
I feel it might be unnecessary to abstract it into a function, so I deleted this function in the latest commit.